contribs/application/src/main/java/org/matsim/application/analysis/CheckPopulation.java

```diff
@@ -164,35 +166,98 @@ public Integer call() throws Exception {
 
 		List<TripStructureUtils.Subtour> subtours = new ArrayList<>();
 
+		NavigableSet<Double> timeGroups = new TreeSet<>(List.of(0d, 1d, 2d, 3d, 4d, 5d, 6d, 7d, 8d, 9d, 10d, 11d,
+			12d, 13d, 14d, 15d, 16d, 17d, 18d, 19d, 20d, 21d, 22d, 23d));
+		Map<String, Double2IntMap> actStartTimes = new HashMap<>();
+		Map<String, Double2IntMap> actEndTimes = new HashMap<>();
+		int noStartTime = 0;
+		int noEndTime = 0;
+		int noStartAndEndTime = 0;
+		int zeroDurationActs = 0;
+		int numberOfActs = 0;
+
 		for (Person agent : agents) {
 
 			// if there are no facility or link ids. the coordinate is used as proxy id.
 			for (PlanElement el : agent.getSelectedPlan().getPlanElements()) {
-				if (el instanceof Activity) {
-					Activity act = (Activity) el;
-					if (act.getFacilityId() == null && act.getLinkId() == null) {
-						act.setLinkId(Id.createLinkId(act.getCoord().toString()));
-					}
+				if (el instanceof Activity act && act.getFacilityId() == null && act.getLinkId() == null) {
+					act.setLinkId(Id.createLinkId(act.getCoord().toString()));
 				}
 			}
 
 			subtours.addAll(TripStructureUtils.getSubtours(agent.getSelectedPlan()));
 
-			List<String> activities = TripStructureUtils.getActivities(agent.getSelectedPlan(), TripStructureUtils.StageActivityHandling.ExcludeStageActivities)
-					.stream().map(CheckPopulation::actName).collect(Collectors.toList());
+			List<Activity> activities = new ArrayList<>(TripStructureUtils.getActivities(agent.getSelectedPlan(), TripStructureUtils.StageActivityHandling.ExcludeStageActivities));
 
-			if (activities.size() == 0)
+			if (activities.isEmpty())
 				continue;
 
-			firstAct.mergeInt(activities.get(0), 1, Integer::sum);
-			lastAct.mergeInt(activities.get(activities.size() - 1), 1, Integer::sum);
-			activities.forEach(act -> acts.mergeInt(act, 1, Integer::sum));
+			firstAct.mergeInt(actName(activities.getFirst()), 1, Integer::sum);
+			lastAct.mergeInt(actName(activities.getLast()), 1, Integer::sum);
+			activities.forEach(act -> acts.mergeInt(actName(act), 1, Integer::sum));
+
+			for (Activity act : Sets.newHashSet(activities)) {
+				String name = actName(act);
+				haveActivity.mergeInt(name, 1, Integer::sum);
+
+				boolean hasNoStartTime = false;
+				boolean hasNoEndTime = false;
+
+//				find time bin for start and end of act or increase counter if no start or end time is defined
+				hasNoStartTime = categorizeTime(act.getStartTime(), timeGroups, actStartTimes, name, hasNoStartTime);
+				hasNoEndTime = categorizeTime(act.getEndTime(), timeGroups, actEndTimes, name, hasNoEndTime);
+
+//				increase counters
+				if (hasNoStartTime || hasNoEndTime) {
+					noStartTime += hasNoStartTime ? 1 : 0;
+					noEndTime += hasNoEndTime ? 1 : 0;
+					noStartAndEndTime += (hasNoStartTime && hasNoEndTime) ? 1 : 0;
+				}
 
-			for (String act : Sets.newHashSet(activities)) {
-				haveActivity.mergeInt(act, 1, Integer::sum);
+				double duration = 0.;
```

Comment:
this should probably use methods from TimeInterpretation, as this is what we _should_ use later in the sim, as well.
See also #3725